clang-tidy-checks/potential-unbounded-loop-check.diff

```diff
@@ -0,0 +1,3051 @@
++      f.write(line.encode())
+ 
+ 
+ # Adds a release notes entry.
+@@ -198,22 +197,22 @@ def add_release_notes(clang_tidy_path, old_check_name, new_check_name):
+ 
+         if match:
+           header_found = True
+-          f.write(line)
++          f.write(line.encode())
+           continue
+ 
+         if line.startswith('^^^^'):
+-          f.write(line)
++          f.write(line.encode())
+           continue
+ 
+         if header_found and add_note_here:
+           if not line.startswith('^^^^'):
+-            f.write("""- The '%s' check was renamed to :doc:`%s
++            f.write(("""- The '%s' check was renamed to :doc:`%s
+   <clang-tidy/checks/%s>`
+ 
+-""" % (old_check_name, new_check_name, new_check_name))
++""" % (old_check_name, new_check_name, new_check_name)).encode())
+             note_added = True
+ 
+-      f.write(line)
++      f.write(line.encode())
+ 
+ def main():
+   parser = argparse.ArgumentParser(description='Rename clang-tidy check.')
+@@ -259,9 +258,9 @@ def main():
+             (check_name_camel, cmake_lists))
+       return 1
+ 
+-    modulecpp = filter(
++    modulecpp = next(filter(
+         lambda p: p.lower() == old_module.lower() + 'tidymodule.cpp',
+-        os.listdir(old_module_path))[0]
++        os.listdir(old_module_path)))
+     deleteMatchingLines(os.path.join(old_module_path, modulecpp),
+                       '\\b' + check_name_camel + '|\\b' + args.old_check_name)
+ 
+diff --git a/clang-tools-extra/docs/ReleaseNotes.rst b/clang-tools-extra/docs/ReleaseNotes.rst
+index 0471c5e9c4eb..733f016682bd 100644
+--- a/clang-tools-extra/docs/ReleaseNotes.rst
++++ b/clang-tools-extra/docs/ReleaseNotes.rst
+@@ -301,6 +301,11 @@ New checks
+   Finds includes of system libc headers not provided by the compiler within
+   llvm-libc implementations.
+ 
++- New :doc:`misc-my-first-check
++  <clang-tidy/checks/misc-potential-unbounded-loop>` check.
++
++  FIXME: add release notes.
++
+ - New :doc:`misc-no-recursion
+   <clang-tidy/checks/misc-no-recursion>` check.
+ 
+diff --git a/clang-tools-extra/docs/clang-tidy/checks/list.rst b/clang-tools-extra/docs/clang-tidy/checks/list.rst
+index 78fed1a8ed9c..b8d9b63bedb9 100644
+--- a/clang-tools-extra/docs/clang-tidy/checks/list.rst
++++ b/clang-tools-extra/docs/clang-tidy/checks/list.rst
+@@ -70,7 +70,7 @@ Clang-Tidy Checks
+    `bugprone-misplaced-widening-cast <bugprone-misplaced-widening-cast.html>`_,
+    `bugprone-move-forwarding-reference <bugprone-move-forwarding-reference.html>`_, "Yes"
+    `bugprone-multiple-statement-macro <bugprone-multiple-statement-macro.html>`_,
+-   `bugprone-no-escape <bugprone-no-escape.html>`_, "Yes"
++   `bugprone-no-escape <bugprone-no-escape.html>`_,
+    `bugprone-not-null-terminated-result <bugprone-not-null-terminated-result.html>`_, "Yes"
+    `bugprone-parent-virtual-call <bugprone-parent-virtual-call.html>`_, "Yes"
+    `bugprone-posix-return <bugprone-posix-return.html>`_, "Yes"
```

Comment:
what caused this change?

Reply:
It's the python script (that came with clang-tidy) to autogenerate boilerplate code. I didn't modify this file manually.